richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -132,7 +132,8 @@ public String substring(int from) {
             int segIdx = pos.getMajor();
             List<SEG> segs = new ArrayList<>(segIdx + 1);
             segs.addAll(segments.subList(0, segIdx));
-            segs.add(segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor()));
+            segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
+                            .ifPresent(seg -> segs.add(seg));
```

Comment:
So, does the way `Paragraph` handle the code need to be changed? Or is the `Optional` approach itself flawed?